File: Retos/Reto #37 - COLORES HEX Y RGB [Media]/python/EspinoLeandroo.py
def rgb_to_hex(r, g, b):
    return "#{:02X}{:02X}{:02X}".format(r, g, b)

def hex_to_rgb(hex_color):
    hex_color = hex_color.lstrip("#")
    r = int(hex_color[0:2], 16)
    g = int(hex_color[2:4], 16)
    b = int(hex_color[4:6], 16)
    return "r: {}, g: {}, b: {}".format(r, g, b)

# rgb_to_hex
r, g, b = 250, 186, 218
hex_color = rgb_to_hex(r, g, b)
print(hex_color)  # Imprime "#000000"

# hex_to_rgb
hex_color = "#fabada"
rgb_color = hex_to_rgb(hex_color)
print(rgb_color)  # Imprime "r: 0, g: 0, b: 0"

File: Retos/Reto #37 - COLORES HEX Y RGB [Media]/java/EspinoLeandroo.java
public class EspinoLeandroo {

    public static String rgbToHex(int r, int g, int b) {
        return String.format("#%02X%02X%02X", r, g, b);
    }    

    public static String hexToRgb(String hex) {
        hex = hex.replace("#", "");
        int bigint = Integer.parseInt(hex, 16);
        int r = (bigint >> 16) & 255;
        int g = (bigint >> 8) & 255;
        int b = bigint & 255;
        return String.format("r: %d, g: %d, b: %d", r, g, b);
    }
    
    public static void main(String[] args) {
        // rgbToHex:
        int r = 250, g = 186, b = 218;
        String hexColor = rgbToHex(r, g, b);
        System.out.println(hexColor);
        // hexToRgb:
        hexColor = "#fabada";
        String rgbColor = hexToRgb(hexColor);
        System.out.println(rgbColor);
    }
}
